### DISCUSSION TITLE
Comma-separated cask versions are problematic in installation paths

### DISCUSSION THREAD

**Discussion by justinkb** [Homebrew/discussions#3024] [Writing Formulae/Casks]
I've just created [an issue in homebrew-cask](https://github.com/Homebrew/homebrew-cask/issues/119618) that shows this design choice has been a poor one.

If we want to prevent other issues that stem from the same problem from occurring (some may already be occurring but have gone unnoticed), what is the best way forward?

**Comment by SMillerDev** [comment]
The best way is to ask upstream to stop using commas in their version. It's not that weird to have two versions of the same program in a directory with a version number to identify them.

**Comment by justinkb** [comment]
Again, as I pointed out in the issue itself, you're completely
misunderstanding the issue, there is no comma in upstream versioning for
Ghidra...

Please, before you comment, verify your claims

**Reply by SMillerDev** [reply]
Okay, let me rephrase it so that it applicable here too.

> The best way is to ask upstream to stop using multiple parts in their version. It's not that weird to have two versions of the same program in a directory with a version number to identify them.

**Reply by justinkb** [reply]
again, you're still missing the point two-fold.

1) there aren't multiple parts to the version, the version is "x.y.z", they just tag the release zip file with the release date, too.
2) the comma in the installation path *is* coming from homebrew, not from Ghidra upstream, it is *contrary to your claim* homebrew's fault this bug happens

**Reply by SMillerDev** [reply]
> there aren't multiple parts to the version, the version is "x.y.z", they just tag the release zip file with the release date, too.

So... that's two then. Since the version clearly isn't unique enough, otherwise they wouldn't include the release date. That date means something to me if I see it. It means `x.y.z that was tagged on <date>`.

> the comma in the installation path is coming from homebrew, not from Ghidra upstream, it is contrary to your claim homebrew's fault this bug happens

Is it Homebrew's fault that the program can't deal with a file path that has valid characters? Because the case you describe would happen if I named my macOS user `some,user` too.

**Comment by justinkb** [comment]
Usernames cannot contain commas

**Reply by neutric** [reply]
File paths can, though, and log4j, as you said in the other thread, "interprets the comma-path as a comma-separated list of more than one path." One can certainly argue that Homebrew, or Apple for that matter, shouldn't allow certain characters for easier interoperability, but one could also argue that log4j is not taking proper precautions for the reality that is macOS.

**Reply by justinkb** [reply]
the comma in the path here is a design *choice* explicitly of homebrew cask, an ill-considered one, it has nothing to do with "realities of macOS" as such. Refer to https://docs.brew.sh/Cask-Cookbook in the end of section Stanza: version.

**Reply by neutric** [reply]
MacOS allows users to create directories with names that have, among others, commas in them. And so they will. Ignoring that is equally, if not more so, ill-considered.
